cg/apps/mip/parse_sampleinfo.py

```diff
@@ -1,71 +1,82 @@
-from typing import List
+from typing import Dict, List, Literal
 
+from pydantic import BaseModel, EmailStr, parse_obj_as
 import yaml
 from cg.apps.tb.models import TrailblazerAnalysis
 
 
-def get_sample_data_from_config(config: dict) -> List[dict]:
+class MipBaseConfig(BaseModel):
+    """This model is used when validating the mip analysis config"""
+
+    analysis_type: Dict[str, str]
+    case_id: str = None
+    config_file_analysis: str
+    email: EmailStr
+    family_id: str = None
+    dry_run_all: bool
+    log_file: str
+    outdata_dir: str
+    slurm_quality_of_service: Literal["low", "normal", "high"]
+    sample_info_file: str
+
+
+class MipBaseSampleinfo(BaseModel):
+    """This model is used when validating the mip samplinfo file"""
+
+    analysisrunstatus: str
+    analysis_date: str
+    case_id: str
+    family_id: str = None
+    human_genome_build: dict
```

Comment:
Same here, expose this even more. I'll be happy tp assist if you have questions how to